apps/api/src/app/shared/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -1,41 +1,156 @@
+  public isClusterMode(): boolean {
+    return process.env.IN_MEMORY_CLUSTER_MODE_ENABLED === 'true';
+  }
+
+  public getClusterOptions(): ClusterOptions | undefined {
+    if (this.inMemoryProviderClient && this.isClusterMode()) {
+      return this.inMemoryProviderClient.options;
+    }
+  }
+
+  public getOptions(): RedisOptions | undefined {
+    if (this.inMemoryProviderClient) {
+      if (this.isClusterMode()) {
+        const clusterOptions: ClusterOptions = this.inMemoryProviderClient.options;
+
+        return clusterOptions.redisOptions;
+      } else {
+        const options: RedisOptions = this.inMemoryProviderClient.options;
+
+        return options;
+      }
+    }
+  }
+
+  private isElasticacheEnabled(): boolean {
+    return !!process.env.ELASTICACHE_CLUSTER_SERVICE_HOST && !!process.env.ELASTICACHE_CLUSTER_SERVICE_PORT;
+  }
+
+  private getClientAndConfigForCluster(): {
+    getClient: () => Cluster | undefined;
+    getConfig: () => InMemoryProviderConfig;
+  } {
+    const clusterProviders = {
+      elasticache: {
+        getClient: getElasticacheCluster,
+        getConfig: getElasticacheClusterProviderConfig,
+      },
+      redis: {
+        getClient: getRedisCluster,
+        getConfig: getRedisClusterProviderConfig,
+      },
+    };
+
+    console.log('hey', this.isElasticacheEnabled());
+
+    return this.isElasticacheEnabled() ? clusterProviders.elasticache : clusterProviders.redis;
+  }
+
+  private inMemoryClusterProviderSetup(): Cluster | undefined {
     Logger.verbose('In-memory cluster service set up', LOG_CONTEXT);
 
-    return {} as InMemoryProviderClient;
+    const { getConfig, getClient } = this.getClientAndConfigForCluster();
+
+    this.inMemoryProviderConfig = getConfig();
+    const { host, ttl } = getConfig();
+
+    if (!host) {
+      Logger.log('Missing host for in-memory cluster provider', LOG_CONTEXT);
+    }
+
+    const inMemoryProviderClient = getClient();
+    if (host && inMemoryProviderClient) {
+      Logger.log(`Connecting to cluster at ${host}`, LOG_CONTEXT);
+
+      inMemoryProviderClient.on('connect', () => {
+        Logger.log('In-memory cluster connected', LOG_CONTEXT);
+      });
+
+      inMemoryProviderClient.on('connecting', () => {
+        Logger.log('In-memory cluster connecting', LOG_CONTEXT);
+      });
+
+      inMemoryProviderClient.on('reconnecting', () => {
+        Logger.log('In-memory cluster reconnecting', LOG_CONTEXT);
+      });
+
+      inMemoryProviderClient.on('close', () => {
+        Logger.warn('In-memory cluster close', LOG_CONTEXT);
+      });
+
+      inMemoryProviderClient.on('end', () => {
+        Logger.warn('In-memory cluster end', LOG_CONTEXT);
+      });
+
+      inMemoryProviderClient.on('error', (error) => {
+        Logger.error(error, LOG_CONTEXT);
+      });
+
+      inMemoryProviderClient.on('ready', () => {
+        Logger.warn('In-memory cluster ready', LOG_CONTEXT);
```

Comment:
```suggestion
        Logger.log('In-memory cluster ready', LOG_CONTEXT);
```

I think better suited as it's a success one

Reply:
I set it up higher as the readiness was being a problem because my infra problems. I can revert it back. 👍🏻